### PR TITLE
Update Nuxt to 2.17 and node to v18.15.0

### LINKED ISSUE #2429: Update `nuxt` to v2.17.0 and node to 18.x.x
## Problem

Nuxt v2.16.x does not support building in node v18, and node v16 is going to be EOL very soon.

## Description

With the new minor release, Nuxt now supports node v18 and v20 🎉 This means that we can update the node version to an LTS even before migrating to Nuxt 3. 
Which node version should we move to, 18.x or 20.x?

## Alternatives

Migrate to Nuxt 3 - but this is a very complex migration.

## Additional context

Nuxt 2.17 release notes: https://github.com/nuxt/nuxt/releases/tag/v2.17.0

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2429 by @obulat

## Description

This PR updates nuxt to version 2.17, and node to version 18.16 🎉 
It also updates some related dependencies.

With this update, the migration to Nuxt 3 will not be as urgent anymore. At least, it won't be tied to Node 16 EOL.

I tried building and running the Docker image locally, and it worked. @sarayourfriend, do we need any changes in the infrastructure to match this change?

## Testing Instructions

I'm not sure how best to test this locally, because I had to completely remove node from my machine to use v18.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — CHANGES_REQUESTED** [review]
LGTM! Great news. I verified that the `FRONTEND_NODE_VERSION` is set correctly by executing the command locally.

```shell

openverse/frontend on  update/nuxt +/- [$] via  v16.14.2
❯ grep "volta" -A 1 ../package.json | tail -n 1 | awk -F'[:"]' '{print $5}' | cut -d. -f1
18
```

The built app runs great locally as well. Anecdotally, I think the local dev server feels a lot faster.

I did find a few places we may need to make additional version updates:

https://github.com/WordPress/openverse/blob/11e44c4b97f2456ba5f0427ecd271e549871a910/.devcontainer/devcontainer.json#L22

And _definitely_ this one:

https://github.com/WordPress/openverse/blob/11e44c4b97f2456ba5f0427ecd271e549871a910/.github/actions/setup-env/action.yml#L62

Not positive about this one:

https://github.com/WordPress/openverse/blob/11e44c4b97f2456ba5f0427ecd271e549871a910/.github/actions/load-img/action.yml#L19

I'll approve this quickly after those action references are fixed.

**Comment by sarayourfriend** [comment]
> @sarayourfriend, do we need any changes in the infrastructure to match this change?

Nope! As long as the image has the same external API (healthcheck behaviour, ports, etc), ECS will deploy whatever image we give it, regardless of what's inside of it.

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
> I'm not sure how best to test this locally, because I had to completely remove node from my machine to use v18.

Are you using Volta to manage Node and pnpm? It's much easier to test this and go back and forth if you're using Volta.

Testing instructions using Volta:

```
# In Openverse project root
$ volta install node
$ volta install pnpm@(just frontend/pnpm-version)
$ just frontend/run i18n
# Note that this logs an error even when it works fine https://github.com/WordPress/openverse/issues/2436
$ just frontend/run prod
# Confirm you can make requests and the app works; Playwright also confirms this
$ just frontend/build-img
$ docker run -p 8443:8443 openverse-frontend:local
# Confirm you can visit the site at http://localhost:8443 and that it works
```

To fix the node version issues Zack pointed out in the remove the hard coded node version and pass `node-version-file: package.json`. [The action knows how to read the volta version](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file) so we don't need to hard code this there.

I couldn't easily find an analogue for the devcontainers though, so for now that one just needs to be updated by hand :cry:

**Comment by obulat** [comment]
> Are you using Volta to manage Node and pnpm? It's much easier to test this and go back and forth if you're using Volta.

Thank you for adding testing instructions, @sarayourfriend! I think I had volta _and_ manually installed node, and they were used one in place of the other.

> To fix the node version issues Zack pointed out in the remove the hard coded node version and pass `node-version-file: package.json`. [The action knows how to read the volta version](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file) so we don't need to hard code this there.

This is neat! Thanks! This will mean we'll need fewer changes when updating to node 20 :)

**Comment by sarayourfriend** [comment]
> I think I had volta and manually installed node, and they were used one in place of the other.

:scream: I had this situation when I first started using Volta and it was so confusing!

**Review by sarayourfriend — APPROVED** [review]
:rocket:

File: .github/actions/load-img/action.yml
Comment (zackkrida) [inline .github/actions/load-img/action.yml:1]: TIL, nice!

**Review by zackkrida — APPROVED** [review]
Awesome 🚀

**Pull request merged**